### ISSUE TITLE
coupon with maximum uses per customers, also for guest checkout (use email)

### ISSUE THREAD

**Issue by seansan** [magento/magento2#776]
When a coupon is made with maximum uses per customers. Then one would expect it to check the database for an order from that customer (and we have the email). Or at least an option: if not logged in, check email too. 

So the option use coupon once only counts logged in customers

When there should also be a check for (unique) email counts per order for guest checkouts

This would be great to add. We have a customer who has ordered every month the last year with a coupon code that can only be used once per customer. It is not closed loop solution: but checking for an already existing order from the email you would solve 99%

http://magento.stackexchange.com/questions/4785/magento-coupon-codes-only-one-coupon-guest-customer

**Comment by tkacheva** [comment]
@seansan 
Currently we support maximum coupon uses per customers only for logged in users

**Comment by seansan** [comment]
this I know

hence the request to create a setting to maximize it per unique email
address

**Comment by tkacheva** [comment]
@seansan
Isn't it very easy to customers to use different emails on checkout to get the promotional discount?

**Comment by seansan** [comment]
Many articles have been written about this

No. They don't. 

But some people do. But they need to know. Need to want to do it. And actually do it. 

The solution is better than now = no functionality to check at all









Sent from my iPhone.

On 28 Jan 2015, at 17:18, tkacheva <notifications@github.com> wrote:

@seansan
Isn't it very easy to customers to use different emails on checkout to get the promotional discount?

**Comment by tkacheva** [comment]
We will look into this. Thank you!

**Comment by vpelipenko** [comment]
Internal ticket: MAGETWO-32177

**Comment by sonalagr** [comment]
I also want to limit guest user to use coupon only once.I also want this feature. Does anyone have the solution?

**Comment by seansan** [comment]
I agree - but think this is a new ticket. With its own argumentation.

**Comment by sonalagr** [comment]
Hi Sensan,
    Can you provide any solution regarding limiting usage for guest user?

**Comment by rex-z** [comment]
@Hi there,
It has been more than a year. Was there any update? Will this feature exist on Magento 2.0?

@sonalagr , for now you will have to set the promo code to be valid for logged in customers only (remove NOT LOGGED IN option).
Another option is to use auto generation code so every code is unique, and make it one time use storewide.

**Comment by tkacheva** [comment]
@rex-z coupon enhancements is a part of promotion scope.  Follow the roadmap updates

**Comment by seansan** [comment]
@tkacheva : "Follow the roadmap updates" 

Where can these be found? (did not know this either)

**Comment by tkacheva** [comment]
@seansan they are given at magento live and imagine conference

**Comment by seansan** [comment]
OK. THX

FYI Would be really good to have this online somewhere.

**Comment by piotrekkaminski** [comment]
Thank you for your submission.

We recently made some changes to the way we process GitHub submissions to more quickly identify and respond to core code issues.

Feature Requests and Improvements should now be submitted to the new [Magento 2 Feature Requests and Improvements](https://community.magento.com/t5/Magento-2-Feature-Requests-and/idb-p/feature-requests) forum (see details [here](https://community.magento.com/t5/News-Announcements/Improvements-to-GitHub-Management/m-p/44572#M96)).

We are closing this GitHub ticket and have moved your request to the new [forum](https://community.magento.com/t5/Magento-2-Feature-Requests-and/coupon-with-maximum-uses-per-customers-also-for-guest-checkout/idi-p/46976).

**Comment by brushbaby** [comment]
Hi There, I have also requested this feature too and be referenced to this thread. I cannot find this specific ticket easily on the new forum. Is anyone able to provide an update on when this feature is likely to be available? We really need to be able to limit a coupon code to a 1 time use only for non logged in customers. Please advise the status/is this going to be introduced and when? Thanks

**Comment by srenon** [comment]
Since this core fix only seem to apply to M2.3 try https://github.com/magepal/coupon-limit-guest-checkout
